modules/openapi-generator/src/main/java/org/openapitools/codegen/DefaultCodegen.java

```diff
@@ -5439,7 +5444,7 @@ public CodegenParameter fromRequestBody(RequestBody body, Set<String> imports, S
                         codegenParameter.description = codegenModelDescription;
                         imports.add(codegenParameter.baseType);
 
-                        if (codegenProperty.complexType != null) {
+                        if (codegenProperty.complexType != null && (useOneOfInterfaces && !templateDir.equals("JavaSpring"))) {
```

Comment:
Same as above, could you please explain this a bit?

Reply:
I wanted here to be sure that, the changes in code will be executed only if it is not spring.
Because this is wrong import, for spring for sure.